src/Rule/BrokenLink.php

```diff
@@ -19,10 +19,15 @@ public function id()
 
 	public function check()
 	{
+		$linkArray = [];
```

Comment:
This is close, but the $linkArray needs to be stored outside of the rule class. This will eliminate duplicates within a single rule, but won't help with duplicates for multiple rules.

Can we put the $linkArray at the PhpAlly class? Maybe a static variable? Then all rules that check links can use it.